### PR TITLE
feat!: Replace ‘on background’ props with ‘inverseColor’ and ‘contrastColor’ for Link, Link List Link, and Icon Button

### PR DESCRIPTION
**Pull request by VincentSmedinga**
# Describe the pull request

Thank you for contributing to the project!
Please use this template to help us handle your PR smoothly.

## What

This replaces `onBackground="light"` with `contrastColor={true}` and `onBackground="dark"` with `inverseColor={true}` for Link, Link List Link, and Icon Button. Related tokens and class names are affected.

## Why

We had an inconsistency between this `onBackground` prop and the `inverseColor` prop of many other components. We’ve analyzed that and found that adding a `constrastColor` prop allows us to use the `inverseColor` prop consistently.

Shortly, the ‘contrast color’ is an alternative foreground colour for a text or icon component if they are on a light (not white) background. The ‘inverse color’ is the same for a dark background.

## How

This is largely a ‘find and replace’ refactoring.

I chose not to add `visited` colours to these components, even thought the tokens for it are available in the `link-appearance` layer.

## Checklist

Before submitting your pull request, please ensure you have done the following. Check each checkmark if you have done so or if it wasn't necessary:

- [x] Add or update unit tests
- [x] Add or update documentation
- [x] Add or update stories
- [x] Add or update exports in index.\* files
- [x] Start the PR title with a Conventional Commit prefix, [as explained here](https://github.com/Amsterdam/design-system/blob/main/documentation/publishing.md?plain=1#L11).

## Additional notes

- Provide links to any related issues or discussions.
- Add a link to the specific story in the feature branch deploy.
- Mention any areas where additional review or feedback is needed.

### CODE REVIEW

File: storybook/src/components/shared/decorators.tsx [hunk 1]
```diff
@@ -5,6 +5,13 @@
...
+/** Applies a light background colour if the `contrastColor` arg is true. */
+export const contrastColorDecorator = (Story: StoryFn, context: StoryContext) => (
+  <div className={context.args['contrastColor'] ? 'ams-docs-light-background' : undefined}>
+    <Story />
+  </div>
+)
+
 /** Applies a dark background colour if the `inverseColor` arg is true. */
 export const inverseColorDecorator = (Story: StoryFn, context: StoryContext) => (
   <div className={context.args['inverseColor'] ? 'ams-docs-dark-background' : undefined}>
```
Comment (alimpens) [inline storybook/src/components/shared/decorators.tsx:17]: Can we increase the specificity here? Having both `contrastColor` and `inverseColor` selected, the components currently show the inverse color on a light background.

File: storybook/src/components/LinkList/LinkList.stories.tsx [hunk 2]
```diff
@@ -6,40 +6,63 @@
...
   component: LinkList,
+} satisfies Meta<typeof LinkList>
+
+export default meta
+
+const linkMeta = {
+  component: LinkList.Link,
+  args: {
+    children: linkList[0],
+    contrastColor: undefined,
```
Comment (alimpens) [inline storybook/src/components/LinkList/LinkList.stories.tsx:25]: Is this necessary? You're already defining these controls in the `LinkStoryTemplate`, no? Same goes for `inverseColor` and `size`

File: storybook/src/components/LinkList/LinkList.stories.tsx [hunk 3]
```diff
@@ -6,40 +6,63 @@
...
-export default meta
+}
 
-const linkMeta = {
-  title: 'Components/Navigation/Link List Link',
-  component: LinkList.Link,
+const LinkStoryTemplate: LinkStory = {
+  args: {
+    children: linkList[0],
+    contrastColor: undefined,
```
Comment (alimpens) [inline storybook/src/components/LinkList/LinkList.stories.tsx:49]: I think you can set this to `false` and remove it from the `argTypes`. Storybook can infer the correct control. Same goes for `inverseColor`

File: storybook/src/components/LinkList/LinkList.stories.tsx [hunk 4]
```diff
@@ -6,40 +6,63 @@
...
+    href: '#',
+    icon: Icons.ChevronRightIcon,
+    inverseColor: undefined,
+    size: undefined,
```
Comment (alimpens) [inline storybook/src/components/LinkList/LinkList.stories.tsx:53]: You don't need to set this if you specify the controls in the `argTypes`

File: storybook/src/components/shared/decorators.tsx (hunk 1 above)
Comment (VincentSmedinga) [reply storybook/src/components/shared/decorators.tsx:17]: The specificity was already in place through source order. The order in which Storybook applies the background decorators didn’t match – now they do.

I’ve also documented not to use both props together and verified that the ‘inverse’ appearance wins if one dows.

File: proprietary/tokens/src/components/ams/link-list.tokens.json [hunk 5]
```diff
@@ -1,40 +1,82 @@
 {
   "ams": {
     "link-list": {
-      "gap": { "value": "{ams.space.sm}" },
+      "gap": {
+        "value": "{ams.space.sm}"
+      },
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply proprietary/tokens/src/components/ams/link-list.tokens.json]: I agree; this change was not intentional. I’ve checked and all values are inline unless they don’t fit the line or have a `comment` property.

**Review by alimpens — APPROVED** [review]

**Pull request merged**